.github/workflows/haskell-build.yml

```diff
@@ -122,6 +129,22 @@ jobs:
     - name: Build projects [build]
       run: cabal build all -j
 
+      - name: Run benchmarks
```

Comment:
TODO: run only if we merge a PR to `main` or if the run was manually dispatched.